pkg/cloud/services/ec2/instances.go

```diff
@@ -188,11 +188,13 @@ func (s *Service) CreateInstance(scope *scope.MachineScope, userData []byte) (*i
 	input.UserData = pointer.StringPtr(base64.StdEncoding.EncodeToString(userData))
 
 	// Set security groups.
-	ids, err := s.GetCoreSecurityGroups(scope)
-	if err != nil {
-		return nil, err
+	if !scope.IsExternallyManaged() {
```

Comment:
Since we are discriminating here https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2383/files#r630312878 seems we can drop this.